src/ripple/core/JobCoro.ipp

```diff
@@ -41,21 +42,45 @@ JobCoro::JobCoro (detail::JobCoro_create_t, JobQueue& jq, JobType type,
 
 inline
 void
-JobCoro::yield () const
+JobCoro::yield() const
 {
     (*yield_)();
 }
 
 inline
 void
-JobCoro::post ()
+JobCoro::post()
 {
+    {
+        std::lock_guard<std::mutex> lk(mutex_run_);
+        running_ = true;
+    }
+
     // sp keeps 'this' alive
     jq_.addJob(type_, name_,
         [this, sp = shared_from_this()](Job&)
         {
-            std::lock_guard<std::mutex> lock (mutex_);
+            auto saved = detail::getLocalValues().release();
+            detail::getLocalValues().reset(&lvs_);
+            std::lock_guard<std::mutex> lock(mutex_);
```

Comment:
That might work but @JoelKatz needs to weigh in on the final version because the concurrency here is tricky